### ISSUE TITLE
URL Encoding bug

### ISSUE THREAD

**Issue by arigon** [pquerna/otp#68]
Hello

if the issuer contains a space, the URL encoder will mix up the URL.

```
key, err := totp.Generate(totp.GenerateOpts{
	Issuer:      "Test Issuer",
	AccountName: "my@email.com",
})
```

Actual result:
`otpauth://totp/Test%20Issuer:my@email.com?algorithm=SHA1&digits=6&issuer=Test+Issuer&period=30&secret=QE2C7JXZB3TY3FBKL6PB7PZXP7UCRPOA`

![otp_issue](https://user-images.githubusercontent.com/11992526/147169623-275495d1-78fc-4ff4-bbbf-16b56135e8df.jpg)

Expected result:
`otpauth://totp/Test%20Issuer:my@email.com?algorithm=SHA1&digits=6&issuer=Test%20Issuer&period=30&secret=QE2C7JXZB3TY3FBKL6PB7PZXP7UCRPOA`

![otp_issue_expected](https://user-images.githubusercontent.com/11992526/147169802-44ca1e3a-2f9d-4a23-9408-70515a25be45.jpg)

**Comment by pquerna** [comment]
So, we are just using standard golang `url.Values{}`  and the `Encode` method.  This uses a `+` as to encode a Space.  This is a valid encoding -- alternatively, using `%20` is also valid. maybe call this upstream https://github.com/golang/go/issues/4013

Do you have a Client that isn't supporting the `+` encoding for spaces?

**Comment by arigon** [comment]
I saw this golang issue as well. The client I made the screenshots with is Google Authenticator. My workaround is to concat the URL and encode each parameter.

**Comment by arigon** [comment]
I will explain the differences of both encodings used by net.URL. Go net.URL has two different encoding functions: `url.PathEscape()` and `url.QueryEscape()`. PathEscape will replace a space with %20, QueryEscape will replace a space with +.
The Path of the OTP URL is escaped by PathEscape (space -> %20) and the parameters are escaped by QueryEscape (space -> +).

I'm building my own URL by concating the user defined values and only use PathEscape().

```
func buildOTPURL(issuer, email, algorithm, digits string, key *otp.Key) string {
	otpURL := "otpauth://totp/"
	otpURL += url.PathEscape(issuer)
	otpURL += ":"
	otpURL += url.PathEscape(email)
	otpURL += "?"
	otpURL += "algorithm=" + algorithm
	otpURL += "&digits=" + digits
	otpURL += "&issuer=" + url.PathEscape(issuer)
	otpURL += "&period=" + fmt.Sprint(key.Period())
	otpURL += "&secret=" + key.Secret()

	return otpURL
}
```

**Comment by pquerna** [comment]
Fixed by #78 and the v1.4 release https://github.com/pquerna/otp/releases/tag/v1.4.0
